pkg/services/sso/keycloak_service_builder.go

```diff
@@ -0,0 +1,59 @@
+package sso
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/keycloak"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/client/redhatsso"
+)
+
+var _ KeycloakServiceBuilderSelector = &keycloakServiceBuilderSelector{}
+var _ KeycloakServiceBuilder = &keycloakServiceBuilder{}
+
+type KeycloakServiceBuilder interface {
+	WithRealmConfig(realmConfig *keycloak.KeycloakRealmConfig) KeycloakServiceBuilder
+	Build() KeycloakService
+}
+
+type keycloakServiceBuilder struct {
+	config      *keycloak.KeycloakConfig
+	realmConfig *keycloak.KeycloakRealmConfig
+}
+
+func (builder *keycloakServiceBuilder) Build() KeycloakService {
+	if builder.config.SelectSSOProvider == keycloak.MAS_SSO {
+		realmConfig := builder.realmConfig
+		if realmConfig == nil {
+			realmConfig = builder.config.KafkaRealm
+		}
+		return newKeycloakService(builder.config, realmConfig)
+	} else {
```

Comment:
can we add it here
if `c.OSDClusterIDPRealm` is passed. it should always initialise `newKeycloakService(builder.config, realmConfig)`

not the redhat sso client.

Reply:
Not sure I get it. You mean that even if `SelectSSOProvider` is `Redhat_SSO`, if `c.OSDClusterIDRealm` is passed we create a MAS_SSO service anyway?

Reply:
yes, irrespective of what is set for "SelectSSOProvider". if `c.OSDClusterIDRealm` passed as realm config, keycloak service should be intiated